### PR TITLE
[AWS Bindings S3] Adds Component Metadata Schema

### PR DESCRIPTION
**Pull request by robertojrojas**
# Description
Adds AWS Bindings S3 Component Metadata Schema.

## Issue reference
this PR will close: #2568

### CODE REVIEW

File: bindings/aws/s3/metadata.yaml [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+    - name: list
+      description: "List blob."
+capabilities: []
+authenticationProfiles:
+  - title: "AccessKey and secretKey"
+    description: "Authenticate using accessKey and secretKey."
+    metadata:
+      - name: accessKey
+        type: string
+        required: false
```
Comment (artursouza) [inline bindings/aws/s3/metadata.yaml:29]: Shouldn't this be required within this profile?

File: bindings/aws/s3/metadata.yaml [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+        sensitive: true
+        description: |
+          ID of the AWS account with appropriate permissions to SNS and SQS.
+          Can be secretKeyRef to use a secret reference.
+        example:  "AKIAIOSFODNN7EXAMPLE"
+        default: ""
+      - name: secretKey
+        type: string
+        required: false
```
Comment (artursouza) [inline bindings/aws/s3/metadata.yaml:38]: Shouldn't this be required within this profile?

File: bindings/aws/s3/metadata.yaml [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+      description: "Create blob."
+    - name: get
+      description: "Get blob."
+    - name: delete
+      description: "Delete blob."
+    - name: list
+      description: "List blob."
+capabilities: []
+authenticationProfiles:
+  - title: "AccessKey and secretKey"
```
Comment (ItalyPaleAle) [inline bindings/aws/s3/metadata.yaml:24]: As mentioned in the other PR, we should create a shared auth profile for these

Comment (robertojrojas) [reply bindings/aws/s3/metadata.yaml:24]: @ItalyPaleAle Yes, while I agree with adding the shared/common auth profile, I think it should be addressed as a separate PR.
I went ahead and created an issue for it: **[Component Metadata Schema] Extend to Support Shared Auth Profile** #2935 
At the time this feature is added the various components, like AWS, etc... can be modified to take advantage of the feature.

Unless there are any other changes to this current PR, I think we should just complete it.

Comment (ItalyPaleAle) [reply bindings/aws/s3/metadata.yaml:24]: Let’s do #2935 first. I’d rather have things done correctly right away and merge things once they’re ready. Adding a new auth profile shouldn’t be a lot of work: you need to add it here: https://github.com/dapr/components-contrib/blob/master/.build-tools/pkg/metadataschema/builtin-authentication-profiles.go

**Review by ItalyPaleAle — CHANGES_REQUESTED** [review]
Waiting on #2935

File: bindings/aws/s3/metadata.yaml [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+  output: true
+  operations:
+    - name: create
+      description: "Create blob."
+    - name: get
+      description: "Get blob."
+    - name: delete
+      description: "Delete blob."
+    - name: list
+      description: "List blob."
```
Comment (ItalyPaleAle) [inline bindings/aws/s3/metadata.yaml:21]: Nit:

```suggestion
    - name: create
      description: "Create blob"
    - name: get
      description: "Get blob"
    - name: delete
      description: "Delete blob"
    - name: list
      description: "List blob"
```

File: bindings/aws/s3/metadata.yaml [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+      example: "AKIAIOSFODNN7EXAMPLE"
+      default: ""
+    - name: secretKey
+      type: string
+      required: true
+      sensitive: true
+      description: |
+        The secret key associated with the access key.
+      example: "wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY"
+      default: ""
```
Comment (ItalyPaleAle) [inline bindings/aws/s3/metadata.yaml:41]: These are included in the built-in authentication profile so need to be removed from here

```suggestion
```

File: bindings/aws/s3/metadata.yaml [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+    description: |
+      The name of the S3 bucket to write to.
+    example: '"bucket"'
+    type: string
+  - name: region
+    required: true
+    description: |
+      The AWS region to the instance. See this page for valid regions:
+      https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/Concepts.RegionsAndAvailabilityZones.html. 
+      Ensure that DynamoDB are available in that region.
```
Comment (ItalyPaleAle) [inline bindings/aws/s3/metadata.yaml:54]: DynamoDB?

File: bindings/aws/s3/metadata.yaml [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+    example: '"true", "false"'
+  - name: encodeBase64
+    description: |
+      Configuration to encode base64 file content before return the content.
+      (In case of opening a file with binary content). true is the only allowed
+      positive value. Other positive variations like "True", "1" are not acceptable.  
+    type: bool
+    default: 'false'
+    example: '"true", "false"'
+  - name: disableSSL
```
[the remark this answers is not in the document]
Comment (ItalyPaleAle) [reply bindings/aws/s3/metadata.yaml:78]: Yes I mean can we rename those properties to `disableTLS` and `insecureTLS`? Maintain the old ones and aliases

File: bindings/aws/s3/metadata.yaml (hunk 6 above)
Comment (ItalyPaleAle) [reply bindings/aws/s3/metadata.yaml:37]: Why do we need DynamoDB for S3?

Comment (robertojrojas) [reply bindings/aws/s3/metadata.yaml:37]: that's a typo. Fixing it.

File: bindings/aws/s3/metadata.yaml (hunk 7 above)
[the remark this answers is not in the document]
Comment (robertojrojas) [reply bindings/aws/s3/metadata.yaml:78]: I'm curious. Will making these change hold this PR?

[the remark this answers is not in the document]
Comment (ItalyPaleAle) [reply bindings/aws/s3/metadata.yaml:78]: No, no! I asked if you can open an issue to track that however please :)

[the remark this answers is not in the document]
Comment (robertojrojas) [reply bindings/aws/s3/metadata.yaml:78]: no problem. I'll open the PR.

[the remark this answers is not in the document]
Comment (robertojrojas) [reply bindings/aws/s3/metadata.yaml:77]: https://github.com/dapr/components-contrib/issues/2983

File: bindings/aws/s3/metadata.yaml [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+    required: true
+    description: |
+      The name of the S3 bucket to write to.
+    example: '"bucket"'
+    type: string
+  - name: region
+    required: true
+    description: |
+      The AWS region to the instance. See this page for valid regions:
+      https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/Concepts.RegionsAndAvailabilityZones.html.
```
Comment (ItalyPaleAle) [inline bindings/aws/s3/metadata.yaml:36]: URLs should go in the `url` property not in the description.

Example:

https://github.com/dapr/components-contrib/blob/21885b306b5f46efb3f89c6ee90f78dec68e5f1b/configuration/redis/metadata.yaml#L87-L89

File: bindings/aws/s3/metadata.yaml [hunk 9]
```diff
@@ -0,0 +1,102 @@
...
+  - name: bucket
+    required: true
+    description: |
+      The name of the S3 bucket to write to.
+    example: '"bucket"'
+    type: string
+  - name: region
+    required: true
+    description: |
+      The AWS region to the instance. See this page for valid regions:
```
Comment (ItalyPaleAle) [inline bindings/aws/s3/metadata.yaml:35]: "AWS region to the instance" sounds odd. Maybe "The AWS region where the bucket exists"?

File: bindings/aws/s3/metadata.yaml [hunk 10]
```diff
@@ -0,0 +1,102 @@
...
+      metadata key on a create request that contains empty data fields.
+    example: '"mydir/my-test-file.txt"'
+    type: string
+  - name: presignTTL
+    required: false
+    description: |
+      To presign an object with a specified time-to-live, use the presignTTL metadata
+      key on a create request, as a Go duration.
+    example: '"2h45m"'
+    type: duration
```
Comment (ItalyPaleAle) [inline bindings/aws/s3/metadata.yaml:102]: According to the source code and the docs, `filePath` and `presignTTL` are keys used in the metadata of requests, not for the component. They should not be listed here.

https://docs.dapr.io/reference/components-reference/supported-bindings/s3/

**Review by ItalyPaleAle — CHANGES_REQUESTED** [review]

File: bindings/aws/s3/metadata.yaml [hunk 11]
```diff
@@ -0,0 +1,122 @@
...
+    example: '"http://localhost:4566"'
+    type: string
+  - name: sessionToken
+    required: false
+    sensitive: true
+    description: |
+      AWS session token to use. A session token is only required if you are using 
+      temporary security credentials.
+    example: '"TOKEN"'
+    type: string
```
[the remark this answers is not in the document]
Comment (artursouza) [reply bindings/aws/s3/metadata.yaml:52]: @robertojrojas

**Review by artursouza — APPROVED** [review]

**Comment by artursouza** [comment]
@ItalyPaleAle Can we merge this?

**Pull request merged**